mayastor/src/lib.rs

```diff
@@ -296,25 +301,13 @@ where
 /// Cleanly exit from the program.
 /// NOTE: cannot be called from a future -> double borrow of executor.
 pub fn mayastor_stop(rc: i32) -> i32 {
-    if rc != 0 {
```

Comment:
This won't work.. the global thread the app framework uses is not set. Its not accessible outside of the app framework either. 

This will cause the shutdown code to crash or trip an assert or something nasty like that. Maybe code that uses the app framework should use a different shutdown callback? (which more or less is what at is was but but simply named differently?)

Reply:
yea, I found out exactly what you described :-) I believe it's fixed now, but let's wait for test results. The main idea of the fix is to avoid mixing shutdown path of new env lib and original rust wrappers in lib.rs. mayastor_stop should be used only for progs started by mayastor_start and mayastor_env_stop only for mayastor::environment::env::MayastorEnvironment::new 🤓